specification/ml/put_job/MlPutJobRequest.ts

```diff
@@ -41,12 +42,14 @@ export interface Request extends RequestBase {
     analysis_limits?: AnalysisLimits
     background_persist_interval: Time
     custom_settings?: CustomSettings
-    data_description?: DataDescription
     daily_model_snapshot_retention_after_days?: long
-    groups?: string[]
+    data_description?: DataDescription
+    datafeed_config?: Datafeed
```

Comment:
This is tricky, in this particular object, the `datafeed_id` and `job_id` are not required as we will default to the top level `job_id` for both.

Reply:
Thanks, I've split off new JobConfig and DatafeedConfig types to enable us to have different attributes for GET vs PUT objects.